pkg/bgpv1/manager/reconcilerv2/neighbor.go

```diff
@@ -27,6 +27,7 @@ type NeighborReconciler struct {
 	SecretStore  store.BGPCPResourceStore[*slim_corev1.Secret]
 	PeerConfig   store.BGPCPResourceStore[*v2alpha1.CiliumBGPPeerConfig]
 	DaemonConfig *option.DaemonConfig
+	metadata     map[uint32]NeighborReconcilerMetadata
```

Comment:
General comment around ASN as key for metadata, should it be instance name instead?

Reply:
This was addressed in a separate PR: https://github.com/cilium/cilium/pull/35207